pkg/portal/transport/admin_api_handler.go

```diff
@@ -44,26 +54,45 @@ func (h *AdminAPIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	cfg, err := h.ConfigResolver.ResolveConfig(appID)
 	if err != nil {
+		h.Logger.WithError(err).Debugf("failed to resolve config")
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
+	hostURL, err := url.Parse(cfg.AppConfig.HTTP.PublicOrigin)
+	if err != nil {
+		h.Logger.WithError(err).Debugf("failed to parse public origin: %v", cfg.AppConfig.HTTP.PublicOrigin)
+		http.Error(w, fmt.Errorf("failed to parse public origin: %v", cfg.AppConfig.HTTP.PublicOrigin).Error(), http.StatusInternalServerError)
+		return
+	}
+
 	authKey, ok := cfg.SecretConfig.LookupData(config.AdminAPIAuthKeyKey).(*config.AdminAPIAuthKey)
 	if !ok {
+		h.Logger.WithError(err).Debugf("failed to look up admin API auth key")
 		http.Error(w, "failed to look up admin API auth key", http.StatusInternalServerError)
 		return
 	}
 
 	endpoint, err := h.EndpointResolver.ResolveEndpoint(appID)
 	if err != nil {
+		h.Logger.WithError(err).Debugf("failed to resolve endpoint")
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
+	h.Logger.WithFields(map[string]interface{}{
+		"appID":    appID,
+		"hostURL":  hostURL.String(),
+		"endpoint": endpoint.String(),
+		"host":     hostURL.Host,
+	}).Debugf("proxy admin API")
+
 	proxy := httputil.ReverseProxy{
 		Director: func(req *http.Request) {
 			req.URL = endpoint
-			req.Host = endpoint.Host
+			// We have to set both to ensure Admin API server see the correct host.
+			req.Host = hostURL.Host
```

Comment:
I don't think this is correct? `AppConfig.HTTP.PublicOrigin` is supposed to reach the main server from Internet, while we want to reach the admin API server from Internet. I suppose this is a cluster specific logic to derive correct endpoint for admin API using app ID? For example: `ADMIN_API_ENDPOINT_TEMPLATE=admin.{{ .AppID}}.authgear.com`.